src/components/PostDetail/PostDetail.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState } from 'react';
+import { customFetch } from '../../customFunctions/customFetch';
+import { apiEndpoints } from '../../configs/apiEndpoints';
+import { CustomFormInput } from '../CustomInputForm/CustomInputForm';
+import styled from 'styled-components';
+import { CustomBackButton } from '../CustomButtons/CustomBackButton/CustomBackButton';
+import { appRoutesList } from '../../configs/appRoutesList';
+import { StyledH1 } from '../PostsPage/PostsPage';
+import { useParams } from 'react-router-dom';
+import { NoDataFound } from '../NoDataFound/NoDataFound';
+
+const StyledTextarea = styled.textarea`
+  font-size: 16px;
+  padding: 15px 10px 5px 5px;
+  display: block;
+  width: 90%;
+  border: none;
+  border-bottom: 1px solid ${(props) => props.theme.colors.grey};
+  margin: 25px 0;
+  font-family: "Poppins";
+  resize: none;
+`;
+
+export const StyledDiv = styled.div`
+  padding: 2% 0;
+`;
+
+type CustomTextareaType = {
+  label: string;
+} & React.TextareaHTMLAttributes<HTMLTextAreaElement>;
+
+export const CustomTextarea = ({
+  label,
+  ...otherProps
+}: CustomTextareaType) => (
+  <StyledDiv>
+    <StyledTextarea {...otherProps}></StyledTextarea>
+    {label && <label>{label}</label>}
+  </StyledDiv>
+);
+
+type PostProps = {
+  id: number;
+  user_id: number;
+  title: string;
+  body: string;
+};
+
+type Post = {
+  post: PostProps;
+};
+
+export const PostDetail = () => {
+  const [postDetail, setPostDetail] = useState<Post>({
+    post: {
+      id: 0,
+      user_id: 0,
+      title: '',
+      body: '',
+    },
+  });
+  const [responseFetchError, setResponseFetchError] = useState(false);
+  const { id } = useParams();
+
+  useEffect(() => {
+    const handleGetResponse = async () => {
+      try {
+        const [data, response] = await customFetch<Post['post']>(
```

Comment:
hele, příjde mi to hodně nahardcoděných věcí v jedné komponentě... 

já bych asi udělal něco, co by udělalo abstrakci nad těma HTTP callama pro POSTy

aka serviceLayer fetch je abstrakce nad HTTP requestama a k tomu bych přidal ještě jednu abstrakci, která bude propojovat, na jaké URL adrese se má jak nakonfigurovat HTTP request

a komponenta by pak byla úplně odproštěna od definovaná typů, stahování configů atd, protože by použila krásné rozhraní funkce, která by se starala o HTTP nastavení a komponeta by jen jednodušše volala funkci a awaitovala data na response

něco ve stylu:
```
// postServices.ts
export const getPosts = () => customFetch<Post['post']>( 
  `${apiEndpoints.POSTS_V2}/${id}`,
  { method: 'GET' }
)


// PostDetail.tsx
import * as postServices from './postServices.ts'

```

vypadá to jako drobnost, ale takhle vlastně propojíš konfigurace a datový typy a další meta nastavení konkrétního HTTP requestu (ne všechny jsou takhle jednoduchý)